pwiz_tools/BiblioSpec/src/BlibMaker.cpp

```diff
@@ -130,6 +131,21 @@ int BlibMaker::parseNextSwitch(int i, int argc, char* argv[])
         exit(0);
     } else if (switchName == 't') {
         scoreLookupMode_ = true;
+    }
+    else if (switchName == 'z') {
+        string charges((++i < argc) ? argv[i] : "");

```

Comment:
This line takes too much thinking to understand what it does. Why not just check i+1 >= argc and if it is throw an error like an argument is required? Unless precursorCharges_ can be empty, but that's the same as not passing -z at all?

Reply:
Fair enough, it should just follow the pattern of the rest of the arglist handling.